### PR TITLE
Add docs templates

### PR DESCRIPTION
**Pull request by evilnick**
Adds notes/templates to make it easier to contribute to docs

Notes!

 - The templates are not .md files on purpose
 - The links won't work until after this is merged :chicken: <=> :egg:

### CODE REVIEW

File: docs/src/_parts/template-howto [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+## See also
+
+- [How to xxx with yyy][]
+- [Get more out of ZZZ][]
+
+> ALWAYS use links by reference for easier maintenance!
+
+<!-- LINKS -->
+
+[getting-started-guide]: /snap/tutorial/getting-started
```
Comment (bschimke95) [inline docs/src/_parts/template-howto:62]: nit: newline here and below.

**Review by bschimke95 — APPROVED** [review]
LGTM

File: docs/src/_parts/template-explanation [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+There is no set format to an explanation document. Good examples consist of
+easy to understand text and are quite frequently accompanied by diagrams.
+
+## Diagrams
+
+You can include a diagram as an image using the usual Markdown format:
+
+![Illustration depicting working on components and clouds][logo]
+
+The documentation also supports  various daigrams-as-code options. We
```
Comment (eaudetcobello) [inline docs/src/_parts/template-explanation:14]:
```suggestion
The documentation also supports various diagrams-as-code options. We
```

File: docs/src/_parts/template-explanation [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+Diagrams like this are processed using the 'kroki' directive:
+
+```{kroki} ../../assets/ck-cluster.puml
+```
+
+## Links
+
+Explanations frequently include links to other documents. In particular, please
+consider adding a section titles 'Further Reading' at the end to collate
```
Comment (eaudetcobello) [inline docs/src/_parts/template-explanation:26]:
```suggestion
consider adding a section titled 'Further Reading' at the end to collate
```

File: docs/src/_parts/template-howto [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+## What you'll need
+
+This guide assumes the following:
+
+- You have root or sudo access to the machine.
+- You have a bootstrapped Canonical Kubernetes cluster (see the [Getting
+  Started][getting-started-guide] guide).
+
+> Proceed with clearly labelled steps. N.B. Use 'imperatives' rather than
+> gerunds ('Enable XXX' rather than 'Enabling XXX', 'Check' rather than
```
Comment (eaudetcobello) [inline docs/src/_parts/template-howto:30]: I learned a new word!

File: docs/src/_parts/template-howto [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+sudo k8s status
+```
+
+> Proceed with steps until the task is complete. Consider whether links to related
+> Guides or resources are appropriate
+
+## See also
+
+- [How to xxx with yyy][]
+- [Get more out of ZZZ][]
```
Comment (eaudetcobello) [inline docs/src/_parts/template-howto:56]:
```suggestion
- [Get more out of zzz][]
```

File: docs/src/_parts/template-reference [hunk 6]
```diff
@@ -0,0 +1,84 @@
+# Reference docs template
+
+> Reference guides are technical descriptions of the machinery.
+
+The reference section is distinct from the How to section in that it provides
+information without demonstrating in any depth how that information is to be used.
+What's the point of that?
+
+Often users may need to check some information - "What release included support
+for Xxxx?" or "What options can I supply to Yyyy".
```
Comment (eaudetcobello) [inline docs/src/_parts/template-reference:10]:
```suggestion
for xxxx?" or "What options can I supply to yyyy". 
```

File: docs/src/_parts/template-reference [hunk 7]
```diff
@@ -0,0 +1,84 @@
+# Reference docs template
+
+> Reference guides are technical descriptions of the machinery.
```
Comment (eaudetcobello) [inline docs/src/_parts/template-reference:3]: Nice

File: docs/src/_parts/template-reference [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+
+Consider an application which deals with many different image formats. A useful
+reference page would list the filetypes supported and maybe a list of common
+file extensions and perhaps a brief decscription:
+
+| FileType | Extensions | Notes                                              |
+|----------|------------|----------------------------------------------------|
+| TIFF     | .tif,.tiff | Tag Image File Format - lossless bitmaps           |
+| JPEG     | .jpg,.jpeg | Compressed photo-quality images                    |
+| PNG      | .png       | Portable Network Graphics -lossless bitmaps        |
```
Comment (eaudetcobello) [inline docs/src/_parts/template-reference:24]:
```suggestion
| PNG      | .png       | Portable Network Graphics - lossless bitmaps        |
```

File: docs/src/_parts/template-reference [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+
+This table gives some reference information. It doesn't try to tell you which
+format to use. It doesn't explain in great detail the differences between them.
+It doesn't guide you how to use them. It is simply a reference. You could use
```
Comment (eaudetcobello) [inline docs/src/_parts/template-reference:28]:
```suggestion
It doesn't guide you on how to use them. It is simply a reference. You could use
```

File: docs/src/_parts/template-reference [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+```
+
+This defines an input from a CSV file.
+
+## Tips
+
+- Remember this is about presenting information.
+- People are unlikely to read the entire page in one go - use headings, lists,
+  tables etc to direct them to the information they are looking for.
+- Resist the temptation to over-explain, but do provide links to documents which do!
```
Comment (eaudetcobello) [inline docs/src/_parts/template-reference:84]: Nice!

File: docs/src/_parts/template-tutorial [hunk 11]
```diff
@@ -0,0 +1,97 @@
+# Tutorial docs template
+
+> Tutorials are explained guides which fulfil the objectives of learning and
```
Comment (eaudetcobello) [inline docs/src/_parts/template-tutorial:3]:
```suggestion
> Tutorials are explained guides which fulfill the objectives of learning and
```

Comment (evilnick) [reply docs/src/_parts/template-tutorial:3]: You really need to turn that US dictionary off :)

Comment (evilnick) [reply docs/src/_parts/template-tutorial:3]: Nope. You really need to turn that US dictionary off ;)

**Pull request merged**